File: create-job.py
import urllib.request
import json

data = {
    "targeting": {
        "titles": ["engineer"],
        "locations": ["United States"]
    }
}

req = urllib.request.Request(
    'http://localhost:3000/api/jobs/apollo',
    data=json.dumps(data).encode('utf-8'),
    headers={'Content-Type': 'application/json'},
    method='POST'
)

try:
    with urllib.request.urlopen(req, timeout=10) as response:
        result = response.read().decode('utf-8')
        print(f'Response: {response.status} - {result}')
except Exception as e:
    print(f'Error: {e}')

File: create-job.js
const http = require('http');

const data = JSON.stringify({
  targeting: {
    titles: ["Engineer"],
    locations: ["United States"],
    organizationNumEmployeesRanges: ["51,100", "101,200"],
    organizationIndustryTagIds: ["5567cd4e7369643b70010000"]
  }
});

const options = {
  hostname: 'localhost',
  port: 3000,
  path: '/api/jobs/apollo',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length
  }
};

const req = http.request(options, (res) => {
  let body = '';
  res.on('data', (chunk) => body += chunk);
  res.on('end', () => {
    console.log('Response:', res.statusCode, body);
  });
});

req.on('error', (e) => {
  console.error('Error:', e.message);
});

req.write(data);
req.end();

console.log('Job creation request sent...');
